raft/node_test.go

```diff
@@ -163,55 +173,6 @@ func TestNodePropose(t *testing.T) {
 	}
 }
 
-// TestNodeReadIndex ensures that node.ReadIndex sends the MsgReadIndex message to the underlying raft.
-// It also ensures that ReadState can be read out through ready chan.
-func TestNodeReadIndex(t *testing.T) {
```

Comment:
Why did you remove this case ?

Reply:
```
commit 0a6a94b5e7570f2876e677831964628e9df94272
Author: Tobias Grieger <tobias.b.grieger@gmail.com>
Date:   Tue Sep 6 17:26:13 2022 +0200

    remove TestNodeReadIndex

    This is tested directly at the level of `RawNode` in
    `TestRawNodeReadIndex`. `*node` is a thin wrapper around `RawNode` so
    this is sufficient.

    The reason to remove the test is that it now incurs data races
    since it's not possible to adjust the `readStates` and `step`
    fields while the node is running, and there is no primitive
    to synchronize with its goroutine. This could all be fixed
    but it's not worth it.

    Signed-off-by: Tobias Grieger <tobias.b.grieger@gmail.com>
```